app/models/asset_field_type.rb

```diff
@@ -119,20 +124,16 @@ def validate_asset_content_type
   end
 
   def style_urls
-    if existing_data.empty?
+    if @existing_data.empty?
       (metadata[:styles].map { |key, value| [key, asset.url(key)] }).to_h
     else
-      existing_data.deep_symbolize_keys[:asset][:style_urls]
+      @existing_data.deep_symbolize_keys[:asset][:style_urls]
     end
   end
 
   def existing_metadata
     metadata.except!(:existing_data)
-
-    unless existing_data.empty?
-      metadata[:path].gsub!(":id", existing_data['asset_field_type_id']) if metadata[:path]
-    end
-
+    metadata[:path].gsub!(":asset_title_slug", asset_title_slug) if metadata[:path]
```

Comment:
@ElliottAYoung you originally created this bit, I believe - will there be any repercussions to removing the `unless`?